### PR TITLE
Add Fleet/Agent 8.5.1 release notes

### LINKED ISSUE #2355: [REQUEST]: Create Fleet/Agent release notes for 8.5.1
### Description

We need to create the release notes manually because they are not being generated for 8.5.x releases.

### Resources

n/a

### Collaboration

The documentation team will investigate the issue and create the initial content.

### Point of contact.

**Main contact:** @dedemorton

### PR DESCRIPTION
**Pull request by dedemorton**
Closes #2355

### CODE REVIEW

File: docs/en/ingest-management/release-notes/release-notes-8.5.asciidoc [hunk 1]
```diff
@@ -13,13 +13,36 @@
...
+=== Bug fixes
+
+{fleet}::
+* Make asset tags space aware {kibana-pull}144066[#144066]
+
+{fleet-server}::
+* No bug fixes for this release
+
+{agent}::
+* No bug fixes for this release
```
Comment (cmacknz) [inline docs/en/ingest-management/release-notes/release-notes-8.5.asciidoc:42]: There are a few changes in 8.5.1 for agent based on the changelog fragments, which look correct: https://github.com/elastic/elastic-agent/tree/8.5/changelog/fragments

Additionally, there is a beats bug fix I think we should highlight in the agent release notes: https://github.com/elastic/beats/pull/33169 

> - Beats will now attempt to recover if a lockfile has not been removed {pull}[33169]

The fix is in Beats but the bug primarily happens when Beats run as part of agent, and the problem solved here has been actively causing Beats run by agent to report as unhealthy for a while.

**Comment by dedemorton** [comment]
@cmacknz I can add the entries based on the changelog fragments, but I'm curious why some of the changes happened before the stack version was updated to 8.5.1. https://github.com/elastic/elastic-agent/commits/0269158ceb20d6b6f20c962193fd5d1ab7ef026a

Also [this commit](https://github.com/elastic/elastic-agent/commit/8f5adae9cb538d828f0f1f03d10606532428ac1a) was merged 6 hours ago. Will it really make 8.5.1?

**Comment by cmacknz** [comment]
> Also [this commit](https://github.com/elastic/elastic-agent/commit/8f5adae9cb538d828f0f1f03d10606532428ac1a) was merged 6 hours ago. Will it really make 8.5.1?

Ah no we need to look at the state of that directory at the commit that is in the 8.5 build candidate:

1. BC link is on the dev issue https://github.com/elastic/dev/issues/2158
2. Elastic agent commit is linked in https://staging.elastic.co/8.5.1-ea0c20bb/summary-8.5.1.html#elastic-agent
3. Changelog fragments in the BC commit: https://github.com/elastic/elastic-agent/tree/0269158ceb20d6b6f20c962193fd5d1ab7ef026a/changelog/fragments

**Comment by cmacknz** [comment]
The set of changelog fragements that exist as of the commit for the build candidate are correct, not the current contents of directory at the head of the 8.5 branch.

File: docs/en/ingest-management/release-notes/release-notes-8.5.asciidoc [hunk 2]
```diff
@@ -13,13 +14,45 @@
...
 
+// begin 8.5.1 relnotes
+
+[[release-notes-8.5.1]]
+== {fleet} and {agent} 8.5.1
+
+Review important information about the {fleet} and {agent} 8.5.1 release.
+
+[discrete]
+[[new-features-8.5.1]]
```
Comment (dedemorton) [inline docs/en/ingest-management/release-notes/release-notes-8.5.asciidoc:33]:
```suggestion
[[enhancements-8.5.1]]
```

File: docs/en/ingest-management/release-notes/release-notes-8.5.asciidoc [hunk 3]
```diff
@@ -13,13 +14,45 @@
...
+== {fleet} and {agent} 8.5.1
+
+Review important information about the {fleet} and {agent} 8.5.1 release.
+
+[discrete]
+[[enhancements-8.5.1]]
+=== Enhancements
+
+{agent}::
+* Improve shutdown logs {agent-pull}1618[#1618]
```
Comment (dedemorton) [inline docs/en/ingest-management/release-notes/release-notes-8.5.asciidoc:37]: This does not seem like a "new feature" so I've put it under enhancements.

**Review by cmacknz — APPROVED** [review]
Thanks!

File: docs/en/ingest-management/release-notes/release-notes-8.5.asciidoc (hunk 3 above)
Comment (cmacknz) [reply docs/en/ingest-management/release-notes/release-notes-8.5.asciidoc:37]: Agreed, it makes more sense here.

**Review by kpollich — APPROVED** [review]
Fleet notes 👍

**Pull request merged**